kntest/pkg/kubetest2/gke/options.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gke
+
+import (
+	"github.com/spf13/cobra"
+
+	"knative.dev/test-infra/pkg/clustermanager/kubetest2"
+)
+
+func addOptions(gkeCmd *cobra.Command, cfg *kubetest2.GKEClusterConfig) {
```

Comment:
Is this essentially duplicating kubetest2 parameters? How decoupled can they be?

 I'm concerned about future changes and how that might affect this tool.

Reply:
These flags are useful for Knative and KubeRun, and here we give the default values that are generally used in the CI flows. If users want to pass extra kubetest2 flags, they could use `--extra-kubetest2-flags` that is supported in the parent `kntest kubetest2` command.

So normally the flags here won't need to be changed unless we want to change the supported flags used for Knative and KubeRun tests, which does not happen quite often.

Reply:
By doing so, we'll effectively allow users to know an internal dependency of our tool (kubetest2) and that will complicate migration to other tools in the future. Could we find a set or parameters that could hide the dependency and introduce kntest as a complete intermediate layer?

Reply:
I got your point. 
The major concern for me on considering kntest as a complete intermediate layer is, every time the test flows need a new flag that is supported by `kubetest2`, like in https://github.com/google/knative-gcp/blob/master/test/e2e-wi-tests.sh#L158, we will need to add it here first, this process will increase the maintenance cost a bit.

Reply:
Ideally, every time kubetest2 accepts a new flag/parameter is because cluster creation/deployment/test execution allows a new functionality and we want to provide it through our own interface. If we don't do that, both tools can become incompatible with each other and it will be extremely hard for us to fix it without affecting customers.
What we want to do is introduce a facade pattern which maps almost 1-to-1 to what kubetest2 does. This map is not exactly direct, otherwise there's no point for the existence of kntest in between, but if we decide to set it we should introduce as much encapsulation as possible. Unless it's strictly necessary to surface kubetest2 parameters in our own interface, I'd push to hide it (even though it might complicate further expansions).

Reply:
My another concern for this is it would be a breaking change to remove the support for passing kubetest2 flags transparently, but what you said it completely reasonable. I will manually update the dependencies of the repos that are impacted by this change.

I have made the corresponding change to remove the `--extra-kubetest2-flags`, please refer to the commit history for better diff check.